core/go/internal/sequencer/common/engine_integration.go

```diff
@@ -0,0 +1,334 @@
+	// whether a lost/late signature would trigger a re-assembly of the transaction ( and any transaction that come after it in the sequencer) or whether we could safely ask the assembly
+	// to post hoc sign an assembly
+
+	// The transaction input data that is the sender's intent to perform the transaction for this ID,
+	// MUST be retrieved from the local database. We cannot process it from the data that is received
+	// over the wire from another node (otherwise that node could "tell us" to do something that no
+	// application locally instructed us to do).
+	// TODO is this still necessary? We are not receiving the PreAssembly from the coordinator. We only get it from the originator's state machine which was initialized from reading the DB
+	// there may be some weird cases where we get a assemble request and we have somehow swapped out the memory record of the preassembly since delegating but that is an edge case and not what we should optimize for
+
+	localTx, err := e.resolveLocalTransaction(ctx, transactionID)
+	if err != nil || localTx.Transaction.Domain != e.domainSmartContract.Domain().Name() || localTx.Transaction.To == nil || *localTx.Transaction.To != e.domainSmartContract.Address() {
+		if err == nil {
+			log.L(ctx).Errorf("transaction %s for invalid domain/address domain=%s (expected=%s) to=%s (expected=%s)",
+				transactionID, localTx.Transaction.Domain, e.domainSmartContract.Domain().Name(), localTx.Transaction.To, e.domainSmartContract.Address())
+		}
+		err := i18n.WrapError(ctx, err, msgs.MsgSequencerAssembleRequestInvalid, transactionID)
+		return nil, err
+	}
+	transaction := &components.PrivateTransaction{
+		ID:          transactionID,
+		Domain:      localTx.Transaction.Domain,
+		Address:     *localTx.Transaction.To,
+		PreAssembly: preAssembly,
+	}
+
+	/*
+	 * Assemble
+	 */
+	log.L(ctx).Debugf("Assembling transaction: %+v", transaction)
+	err = e.domainSmartContract.AssembleTransaction(domainContext, e.components.Persistence().NOTX(), transaction, localTx)
+	if err != nil {
+		log.L(ctx).Errorf("error assembling transaction: %s", err)
+		return nil, err
+	}
+	if transaction.PostAssembly == nil {
+		// This is most likely a programming error in the domain
+		err := i18n.NewError(ctx, msgs.MsgSequencerInternalError, "AssembleTransaction returned nil PostAssembly")
+		return nil, err
+	}
+
+	// Some validation that we are confident we can execute the given attestation plan
+	for _, attRequest := range transaction.PostAssembly.AttestationPlan {
+		switch attRequest.AttestationType {
+		case prototk.AttestationType_ENDORSE:
+		case prototk.AttestationType_SIGN:
+		case prototk.AttestationType_GENERATE_PROOF:
+			errorMessage := "AttestationType_GENERATE_PROOF is not implemented yet"
+			log.L(ctx).Error(errorMessage)
+			return nil, i18n.NewError(ctx, msgs.MsgSequencerInternalError, errorMessage)
```

Comment:
Is this a functional regression in the branch at this point in time?

e.g. do we not support domains like Zeto?